Versions update

jklukas opened the pull request:
Support for hyphen-separated key names was dropped in sbt 1.0.0;
see https://www.scala-sbt.org/1.x/docs/sbt-1.0-Release-Notes.html#sbt+1.0.0

Also updates to the Spark version used by current default EMR in ATMO.

jklukas commented:
Travis run is failing. It looks like it's either a dependency issue or a problem accessing the Hive metastore.

jklukas commented:
It appears that the Spark update has pulled in a newer version of derby and I think we're getting hit by https://issues.apache.org/jira/browse/DERBY-6648

Derby implemented new security restrictions that require some additional configuration if running under a SecurityManager. I'm not aware why this code would be running under a SecurityManager, though.

jklukas commented:
> I'm not aware why this code would be running under a SecurityManager, though.

Turns out sbt imposes a SecurityManager in order to help it deal with graceful shutdown. Setting `fork=true` avoids this and may allow our tests to run without tripping the derby security issue.

jklukas commented:
When running the full test suite, looks like `fork=true` is problematic. Tests now fail with things like:

```
ERROR XSDB6: Another instance of Derby may have already booted the database /telemetry-batch-view/metastore_db
```

jklukas commented:
This is finally passing tests and I'm comfortable with the content.

@acmiyaguchi - Can you read through again and let me know if your approval still stands? Does this look good to merge?

fbertsch commented on src/main/scala/com/mozilla/telemetry/views/GenericLongitudinal.scala:
```diff
@@ -84,7 +84,7 @@ object GenericLongitudinalView {
   }
 
-  def getInputTable(sqlContext: SQLContext, opts: Opts): DataFrame = {
+  def getInputTable(sqlContext: SparkSession, opts: Opts): DataFrame = {
```
This variable should be renamed, since it's no longer a `sqlContext`

fbertsch commented on src/main/scala/com/mozilla/telemetry/views/GenericLongitudinal.scala:
```diff
@@ -4,7 +4,7 @@ import com.github.nscala_time.time.Imports._
 import org.apache.spark.sql.functions._
 import org.apache.spark.sql.types.StructType
-import org.apache.spark.sql.{DataFrame, SQLContext}
+import org.apache.spark.sql.{DataFrame, SQLContext, SparkSession}
```
It doesn't look like we need the `SQLContext` import any longer

jklukas commented:
I had a tab sitting open that didn't show @fbertsch's comments yet, so merged before seeing them. I can either back this out for now or I can proceed with running tests today and back out if we see any problems.

acmiyaguchi commented:
@fbertsch What testing needs to be done in staging for a change like this? Is it just verifying that the jobs will run without a fatal exception or an empty output at each end state?

fbertsch commented:
@acmiyaguchi It's not completely clear. There are probably levels of testing:
Basic: Test that a single job starts successfully
Better: Test that a single job runs to completion with full output
Even Better: Test that a variety of jobs run to completion with full output
Best: Test all DAGs, checking that no jobs are affected

"Best" is probably too much, and would be a lot of work and overhead. I think at least the "Better" option should be done, and certainly on at least `main_summary`.

jklukas commented:
I was able to run the full main_summary DAG on nightly data via local Airflow without exceptions. I ran DatasetComparator on that output versus the same day's real data and got the following:

```
-bash-4.2$ spark-submit --class com.mozilla.telemetry.utils.DatasetComparator --master=spark://ec2-34-209-238-34.us-west-2.compute.amazonaws.com telemetry-batch-view-1.1.jar --date=20180523
Picked up _JAVA_OPTIONS: -Djava.io.tmpdir=/mnt1/ -Xmx15000M -Xms1000M
Picked up _JAVA_OPTIONS: -Djava.io.tmpdir=/mnt1/ -Xmx15000M -Xms1000M
Exception in thread "main" org.apache.spark.SparkException: Invalid master URL: spark://ec2-34-209-238-34.us-west-2.compute.amazonaws.com
Null values in 'is_default_browser' went from 682449 to 624
Null values in 'scalar_parent_webrtc_nicer_stun_retransmits' went from 329823769 to 175702
Null values in 'histogram_parent_a11y_consumers' went from 309683247 to 167668
Null values in 'user_pref_dom_ipc_processcount' went from 329467045 to 170699
Null values in 'scalar_gpu_telemetry_discarded_keyed_scalar_actions' went from 118959878 to 46864
Null values in 'histogram_parent_ssl_handshake_version' went from 63571838 to 6492
Null values in 'histogram_content_gc_max_pause_ms_2' went from 76078982 to 15742
Null values in 'plugin_hangs' went from 329680426 to 175889
Null values in 'histogram_parent_memory_heap_allocated' went from 321009524 to 1684
Null values in 'previous_build_id' went from 307048868 to 99763
Null values in 'devtools_toolbox_opened_count' went from 327757080 to 170368
Null values in 'vendor' went from 0 to 0
Null values in 'scalar_parent_browser_engagement_max_concurrent_tab_count' went from 20212060 to 2906
Null values in 'scalar_parent_screenshots_copy' went from 329804205 to 175757
Null values in 'shutdown_kill' went from 319980980 to 171414
Null values in 'crash_submit_attempt_main' went from 329460038 to 175587
Null values in 'histogram_content_http_channel_disposition' went from 329837689 to 175935
Null values in 'histogram_content_time_to_dom_complete_ms' went from 329443641 to 170160
Null values in 'user_pref_dom_ipc_plugins_sandbox_level_flash' went from 329837469 to 175900
Null values in 'histogram_parent_webext_browseraction_popup_open_ms' went from 326596592 to 166201
Null values in 'histogram_parent_webvr_time_spent_viewing_in_oculus' went from 329837676 to 175935
Null values in 'scalar_parent_browser_session_restore_browser_startup_page' went from 94819395 to 167707
Null values in 'cycle_collector_max_pause_main_above_150' went from 0 to 0
Null values in 'scalar_parent_devtools_onboarding_is_devtools_user' went from 329792676 to 174862
Null values in 'histogram_parent_webext_storage_local_set_ms' went from 324047342 to 164282
Null values in 'histogram_parent_memory_unique' went from 321766044 to 2452
Null values in 'env_build_id' went from 1 to 0
Null values in 'scalar_gpu_telemetry_accumulate_unknown_histogram_keys' went from 329837689 to 175935
Null values in 'session_length' went from 109242 to 366
Null values in 'user_pref_browser_search_widget_innavbar' went from 63483157 to 7139
Null values in 'histogram_parent_memory_distribution_among_content' went from 324658585 to 31172
Null values in 'histogram_parent_devtools_entry_point' went from 328077797 to 171235
Null values in 'histogram_content_http_pageload_is_ssl' went from 329837688 to 175935
Null values in 'profile_subsession_counter' went from 1 to 0
Null values in 'scalar_parent_devtools_aboutdevtools_opened' went from 329835666 to 175930
Null values in 'sync_configured' went from 265768025 to 131902
Null values in 'scalar_parent_formautofill_creditCards_fill_type_autofill_modified' went from 329837663 to 175912
Null values in 'histogram_parent_time_to_response_start_ms' went from 329770301 to 175763
Null values in 'scalar_parent_browser_timings_last_shutdown' went from 324517835 to 45789
Null values in 'default_search_engine_data_name' went from 549743 to 639
Null values in 'scalar_parent_devtools_toolbar_eyedropper_opened' went from 329829993 to 175882
Null values in 'scalar_parent_devtools_accessibility_picker_used_count' went from 329837679 to 175934
Null values in 'scalar_content_gfx_small_paint_phase_weight' went from 324213428 to 154547
Null values in 'histogram_parent_cycle_collector_max_pause' went from 68873332 to 9240
Null values in 'scalar_parent_browser_engagement_unfiltered_uri_count' went from 76274609 to 36585
Null values in 'events' went from 324364034 to 87172
Null values in 'scalar_parent_preferences_use_bookmark' went from 329837689 to 175935
Null values in 'previous_subsession_id' went from 1925810 to 956
Null values in 'flash_version' went from 118695832 to 99544
Null values in 'scalar_parent_idb_type_temporary_count' went from 329837689 to 175935
Null values in 'histogram_content_time_to_non_blank_paint_ms' went from 329277374 to 169237
Null values in 'histogram_parent_time_to_first_interaction_ms' went from 329665668 to 175207
Null values in 'histogram_parent_time_to_non_blank_paint_ms' went from 329416556 to 175107
Null values in 'histogram_content_time_to_dom_interactive_ms' went from 329429318 to 170067
Null values in 'histogram_content_fx_tab_close_time_anim_ms' went from 329837689 to 175935
Null values in 'scalar_content_webrtc_nicer_turn_438s' went from 329331576 to 174330
Null values in 'scalar_content_sw_cors_res_for_so_req_count' went from 329837689 to 175935
Null values in 'histogram_content_time_to_first_mouse_move_ms' went from 329609299 to 173592
Null values in 'gfx_features_advanced_layers_status' went from 100731736 to 42466
Null values in 'histogram_content_tracking_protection_enabled' went from 329837689 to 175935
Null values in 'scalar_parent_devtools_toolbox_tabs_reordered' went from 329837420 to 175915
Null values in 'scalar_parent_browser_usage_plugin_instantiated' went from 329717035 to 175918
Null values in 'scalar_content_telemetry_discarded_accumulations' went from 62372675 to 5811
Datasets are not the same
```

That seems vaguely reasonable, with the full dataset results at about a 10000:1 ratio to nightly, but it's not a very nuanced validation.

I may need to run the DAG again using master branch of t-b-v to see if I truly get identical output.

jklukas commented:
I ran the DAG again using current master branch of t-b-v on nightly and compared:

```
-bash-4.2$ spark-submit --class com.mozilla.telemetry.utils.DatasetComparator telemetry-batch-view-1.1.jar --date=20180523 --original-bucket=telemetry-test-bucket --test-bucket=jklukas-test
Picked up _JAVA_OPTIONS: -Djava.io.tmpdir=/mnt1/ -Xmx15000M -Xms1000M
Picked up _JAVA_OPTIONS: -Djava.io.tmpdir=/mnt1/ -Xmx15000M -Xms1000M
Common columns: histogram_content_webext_storage_local_get_ms,scalar_parent_browser_session_restore_worker_restart_count,normalized_channel,scalar_content_telemetry_discarded_keyed_scalar_actions,profile_reset_date,histogram_content_fx_urlbar_selected_result_type,cpu_cores,scalar_content_telemetry_event_counts,plugins_infobar_dismissed,user_pref_extensions_legacy_enabled,crashes_detected_content,histogram_parent_a11y_instantiated_flag,scalar_content_idb_type_persistent_count,gc_max_pause_ms_main_above_2500,histogram_content_input_event_response_coalesced_ms,input_event_response_coalesced_ms_content_above_150,scalar_content_mediarecorder_recording_count,histogram_parent_webvr_time_spent_viewing_in_2d,scalar_parent_security_pkcs11_modules_loaded,active_experiment_id,scalar_parent_devtools_tool_registered,scalar_parent_browser_engagement_window_open_event_count,scalar_parent_devtools_grid_gridinspector_opened,sync_count_mobile,city,histogram_parent_http_pageload_is_ssl,experiments,scalar_gpu_browser_usage_graphite,histogram_content_cert_validation_success_by_ca,cycle_collector_max_pause_main_above_250,env_build_version,previous_session_id,os_service_pack_major,histogram_content_ssl_tls13_intolerance_reason_pre,total_time,histogram_content_fx_urlbar_selected_result_method,places_pages_count,apple_model_id,histogram_content_webext_storage_local_set_ms,histogram_parent_webext_background_page_load_ms,gc_max_pause_ms_main_above_250,scalar_parent_storage_sync_api_usage_extensions_using,histogram_content_uptake_remote_content_result_1,histogram_parent_fx_tab_close_time_anim_ms,firewall,channel,histogram_parent_webext_pageaction_popup_open_ms,histogram_content_time_to_first_click_ms,timezone_offset,histogram_content_time_to_first_interaction_ms,default_search_engine_data_origin,cycle_collector_max_pause_content_above_2500,scalar_parent_mediarecorder_recording_count,update_enabled,cycle_collector_max_pause_content_above_150,scalar_parent_browser_errors_reported_failure_count,session_id,histogram_content_time_to_load_event_start_ms,scalar_parent_sw_synthesized_res_count,scalar_parent_devtools_responsive_open_trigger,histogram_parent_ssl_handshake_result,scalar_parent_browser_engagement_total_uri_count,timestamp,scalar_parent_browser_errors_collected_count_by_filename,histogram_parent_ssl_tls12_intolerance_reason_pre,scalar_parent_devtools_aboutdevtools_installed,default_search_engine_data_load_path,crash_submit_success_content,scalar_parent_dom_contentprocess_buildID_mismatch,keyed_uint_addon_scalars,histogram_gpu_uptake_remote_content_result_1,histogram_parent_http_channel_disposition,default_search_engine_data_submission_url,client_clock_skew,histogram_parent_gc_max_pause_ms_2,sample_id,e10s_enabled,histogram_parent_fx_urlbar_selected_result_type,search_counts,os_version,scalar_parent_browser_feeds_feed_subscribed,popup_notification_stats,crash_submit_success_plugin,scalar_parent_browser_feeds_livebookmark_item_opened,session_start_date,scalar_parent_formautofill_availability,is_wow64,histogram_parent_search_reset_result,histogram_content_memory_vsize_max_contiguous,scalar_parent_devtools_rules_gridinspector_opened,user_pref_extensions_allow_non_mpc_extensions,app_version,windows_ubr,user_pref_general_config_filename,scalar_gpu_telemetry_discarded_accumulations,scalar_parent_browser_engagement_navigation_urlbar,input_event_response_coalesced_ms_content_above_250,histogram_parent_memory_total,scalar_parent_browser_session_restore_number_of_tabs,histogram_parent_cert_validation_success_by_ca,histogram_content_http_transaction_is_ssl,plugins_infobar_block,scalar_parent_services_sync_sync_login_state_transitions,scalar_parent_timestamps_first_paint,plugins_infobar_allow,document_id,scalar_parent_a11y_indicator_acted_on,profile_creation_date,histogram_content_search_reset_result,active_addons_count,uint_addon_scalars,histogram_parent_time_to_first_mouse_move_ms,scalar_parent_navigator_storage_persist_count,scalar_content_telemetry_accumulate_unknown_histogram_keys,scalar_content_browser_usage_graphite,scalar_parent_browser_engagement_navigation_contextmenu,crash_submit_attempt_content,scalar_parent_browser_engagement_active_ticks,subsession_id,scalar_content_telemetry_discarded_scalar_actions,histogram_content_memory_resident_fast,gfx_compositor,scalar_parent_idb_type_persistent_count,ssl_handshake_result,histogram_content_ssl_tls12_intolerance_reason_pre,scalar_content_sw_synthesized_res_count,user_prefs,scalar_content_webrtc_nicer_turn_401s,scalar_parent_preferences_search_query,scalar_parent_browser_engagement_tab_open_event_count,histogram_parent_memory_vsize_max_contiguous,plugins_infobar_shown,scalar_parent_formautofill_addresses_fill_type_autofill,histogram_content_fx_new_window_ms,histogram_parent_devtools_toolbox_time_active_seconds,antispyware,histogram_parent_webext_storage_local_get_ms,cpu_count,scalar_content_telemetry_discarded_child_events,scalar_parent_browser_engagement_restored_pinned_tabs_count,histogram_parent_input_event_response_coalesced_ms,scalar_content_sw_alternative_body_used_count,scalar_parent_browser_engagement_navigation_about_home,plugins_notification_user_action,input_event_response_coalesced_ms_content_above_2500,histogram_content_time_to_first_scroll_ms,os,places_bookmarks_count,scalar_content_gfx_omtp_paint_wait_ratio,scalar_parent_devtools_accessibility_select_accessible_for_node,histogram_parent_webext_extension_startup_ms,scalar_content_telemetry_discarded_keyed_accumulations,cpu_speed_mhz,crashes_detected_plugin,histogram_content_fx_urlbar_selected_result_index_by_type,plugins_notification_shown,scalar_parent_webrtc_nicer_turn_438s,scalar_parent_devtools_responsive_toolbox_opened_first,scalar_parent_browser_session_restore_browser_tabs_restorebutton,client_submission_date,sync_count_desktop,histogram_content_memory_total,scalar_parent_formautofill_creditCards_fill_type_manual,crashes_detected_gmplugin,scalar_parent_telemetry_event_counts,subsession_start_date,scalar_parent_telemetry_accumulate_unknown_histogram_keys,scalar_parent_devtools_grid_showGridAreasOverlay_checked,histogram_parent_webext_content_script_injection_ms,disabled_addons_ids,histogram_parent_tracking_protection_enabled,scalar_parent_formautofill_addresses_fill_type_autofill_update,scalar_parent_preferences_prefs_file_was_invalid,histogram_content_memory_heap_allocated,histogram_parent_uptake_remote_content_result_1,histogram_parent_time_to_first_key_input_ms,cycle_collector_max_pause_content_above_250,scalar_parent_browser_engagement_navigation_searchbar,country,os_service_pack_minor,env_build_arch,histogram_content_devtools_toolbox_time_active_seconds,histogram_content_cycle_collector_max_pause,scalar_parent_devtools_copy_xpath_opened,windows_build_number,scalar_parent_preferences_browser_home_page_count,histogram_content_memory_distribution_among_content,scalar_parent_browser_engagement_navigation_about_newtab,ssl_handshake_result_failure,main,scalar_parent_devtools_accessibility_service_enabled_count,scalar_parent_screenshots_download,histogram_parent_time_to_first_scroll_ms,client_id,gc_max_pause_ms_content_above_150,histogram_parent_webvr_users_view_in,reason,keyed_boolean_addon_scalars,scalar_parent_browser_engagement_unique_domains_count,histogram_content_webvr_time_spent_viewing_in_2d,active_theme,histogram_content_time_to_response_start_ms,histogram_content_ssl_handshake_result,scalar_parent_security_webauthn_used,aborts_plugin,app_display_version,histogram_parent_memory_vsize,attribution,histogram_gpu_webvr_time_spent_viewing_in_openvr,scalar_parent_browser_session_restore_number_of_win,scalar_content_navigator_storage_estimate_count,scalar_parent_screenshots_upload,histogram_parent_ssl_tls13_intolerance_reason_pre,scalar_parent_sandbox_no_job,scalar_parent_gfx_advanced_layers_failure_id,locale,scalar_parent_preferences_use_current_page,push_api_notify,histogram_content_time_to_dom_content_loaded_start_ms,scalar_parent_browser_errors_reported_success_count,histogram_content_time_to_load_event_end_ms,scalar_parent_devtools_accessibility_opened_count,scalar_content_webrtc_nicer_stun_retransmits,scalar_parent_browser_feeds_preview_loaded,boolean_addon_scalars,creation_date,scalar_parent_dom_contentprocess_troubled_due_to_memory,scalar_parent_aushelper_websense_reg_version,histogram_gpu_webvr_time_spent_viewing_in_oculus,scalar_parent_telemetry_accumulate_clamped_values,histogram_content_ssl_handshake_version,scalar_parent_browser_engagement_max_concurrent_window_count,gfx_features_gpu_process_status,search_cohort,histogram_content_webext_content_script_injection_ms,update_channel,scalar_parent_a11y_instantiators,cpu_stepping,scalar_content_webrtc_nicer_turn_403s,scalar_content_idb_type_temporary_count,keyed_string_addon_scalars,first_paint,scalar_content_browser_feeds_preview_loaded,distribution_id,geo_subdivision2,histogram_parent_touch_enabled_device,histogram_content_webvr_users_view_in,cpu_model,sandbox_effective_content_process_level,scalar_parent_browser_session_restore_tabbar_restore_available,subsession_counter,scalar_parent_browser_errors_collected_with_stack_count,input_event_response_coalesced_ms_main_above_2500,cpu_family,update_auto_download,scalar_parent_webrtc_nicer_turn_401s,gfx_features_d3d11_status,install_year,scalar_parent_extensions_updates_rdf,scalar_parent_devtools_grid_showGridLineNumbers_checked,scalar_parent_preferences_created_new_user_prefs_file,scalar_parent_browser_feeds_livebookmark_opened,input_event_response_coalesced_ms_main_above_150,histogram_content_memory_vsize,ssl_handshake_result_success,gc_max_pause_ms_main_above_150,histogram_content_time_to_first_key_input_ms,histogram_content_fx_searchbar_selected_result_method,session_restored,scalar_parent_browser_session_restore_tabbar_restore_clicked,scalar_content_browser_usage_plugin_instantiated,histogram_parent_time_to_first_click_ms,histogram_gpu_webvr_users_view_in,histogram_parent_fx_tab_switch_total_e10s_ms,cycle_collector_max_pause_main_above_2500,histogram_parent_ghost_windows,aborts_gmplugin,scalar_parent_navigator_storage_estimate_count,histogram_content_ghost_windows,e10s_multi_processes,histogram_parent_fx_urlbar_selected_result_index_by_type,geo_subdivision1,histogram_parent_webext_browseraction_popup_preload_result_count,histogram_content_memory_unique,memory_mb,ghost_windows_main_above_1,subsession_length,cpu_vendor,active_addons,crash_submit_attempt_plugin,scalar_parent_devtools_inspector_three_pane_enabled,histogram_content_time_to_dom_loading_ms,scalar_content_mathml_doc_count,scalar_parent_browser_errors_collected_count,normalized_os_version,histogram_parent_fx_urlbar_selected_result_method,scalar_parent_formautofill_addresses_fill_type_manual,histogram_parent_fx_searchbar_selected_result_method,histogram_parent_memory_resident_fast,scalar_gpu_telemetry_discarded_scalar_actions,scalar_parent_devtools_copy_unique_css_selector_opened,histogram_parent_fx_urlbar_selected_result_index,scalar_parent_security_client_cert,histogram_parent_fx_new_window_ms,scalar_parent_sw_alternative_body_used_count,gc_max_pause_ms_content_above_2500,scalar_parent_preferences_browser_home_page_change,scalar_gpu_telemetry_event_counts,active_ticks,scalar_parent_devtools_current_theme,scalar_parent_browser_errors_sample_rate,gfx_features_d2d_status,string_addon_scalars,scalar_parent_devtools_copy_full_css_selector_opened,telemetry_enabled,histogram_content_time_to_dom_content_loaded_end_ms,app_name,histogram_parent_http_transaction_is_ssl,scalar_parent_preferences_prevent_accessibility_services,antivirus,scalar_parent_devtools_accessibility_node_inspected_count,scalar_content_navigator_storage_persist_count,scalar_content_memoryreporter_max_ghost_windows,histogram_parent_webvr_time_spent_viewing_in_openvr,gc_max_pause_ms_content_above_250,blocklist_enabled,scalar_parent_webrtc_nicer_turn_403s,scalar_parent_browser_feeds_livebookmark_count,client_submission_latency,web_notification_shown,quantum_ready,ghost_windows_content_above_1,input_event_response_coalesced_ms_main_above_250,scalar_parent_formautofill_creditCards_fill_type_autofill,active_experiment_branch,scalar_parent_storage_sync_api_usage_items_stored,scalar_parent_storage_sync_api_usage_storage_consumed,scalar_parent_devtools_grid_showInfiniteLines_checked,scalar_parent_devtools_aboutdevtools_noinstall_exits,scalar_parent_sw_cors_res_for_so_req_count,addon_compatibility_check_enabled,aborts_content,scalar_parent_browser_usage_graphite,app_build_id,scalar_parent_preferences_read_user_js,cpu_l3_cache_kb,submission_date,scalar_gpu_telemetry_discarded_child_events,crash_submit_success_main,scalar_gpu_telemetry_discarded_keyed_accumulations,histogram_content_fx_urlbar_selected_result_index,cpu_l2_cache_kb,default_search_engine,is_default_browser,scalar_parent_webrtc_nicer_stun_retransmits,histogram_parent_a11y_consumers,user_pref_dom_ipc_processcount,scalar_gpu_telemetry_discarded_keyed_scalar_actions,histogram_parent_ssl_handshake_version,histogram_content_gc_max_pause_ms_2,plugin_hangs,histogram_parent_memory_heap_allocated,previous_build_id,devtools_toolbox_opened_count,vendor,scalar_parent_browser_engagement_max_concurrent_tab_count,scalar_parent_screenshots_copy,shutdown_kill,crash_submit_attempt_main,histogram_content_http_channel_disposition,histogram_content_time_to_dom_complete_ms,user_pref_dom_ipc_plugins_sandbox_level_flash,histogram_parent_webext_browseraction_popup_open_ms,histogram_parent_webvr_time_spent_viewing_in_oculus,scalar_parent_browser_session_restore_browser_startup_page,cycle_collector_max_pause_main_above_150,scalar_parent_devtools_onboarding_is_devtools_user,histogram_parent_webext_storage_local_set_ms,histogram_parent_memory_unique,env_build_id,scalar_gpu_telemetry_accumulate_unknown_histogram_keys,session_length,user_pref_browser_search_widget_innavbar,histogram_parent_memory_distribution_among_content,histogram_parent_devtools_entry_point,histogram_content_http_pageload_is_ssl,profile_subsession_counter,scalar_parent_devtools_aboutdevtools_opened,sync_configured,scalar_parent_formautofill_creditCards_fill_type_autofill_modified,histogram_parent_time_to_response_start_ms,scalar_parent_browser_timings_last_shutdown,default_search_engine_data_name,scalar_parent_devtools_toolbar_eyedropper_opened,scalar_parent_devtools_accessibility_picker_used_count,scalar_content_gfx_small_paint_phase_weight,histogram_parent_cycle_collector_max_pause,scalar_parent_browser_engagement_unfiltered_uri_count,events,scalar_parent_preferences_use_bookmark,previous_subsession_id,flash_version,scalar_parent_idb_type_temporary_count,histogram_content_time_to_non_blank_paint_ms,histogram_parent_time_to_first_interaction_ms,histogram_parent_time_to_non_blank_paint_ms,histogram_content_time_to_dom_interactive_ms,histogram_content_fx_tab_close_time_anim_ms,scalar_content_webrtc_nicer_turn_438s,scalar_content_sw_cors_res_for_so_req_count,histogram_content_time_to_first_mouse_move_ms,gfx_features_advanced_layers_status,histogram_content_tracking_protection_enabled,scalar_parent_devtools_toolbox_tabs_reordered,scalar_parent_browser_usage_plugin_instantiated,scalar_content_telemetry_discarded_accumulations
Added columns:
Dropped columns: scalar_parent_telemetry_persistence_timer_hit_count,scalar_content_telemetry_profile_directory_scans,scalar_parent_telemetry_profile_directory_scans,scalar_gpu_telemetry_profile_directory_scans
Row count went from 175935 to 175935
Added 0 row values
Dropped 0 row values
Null values in 'histogram_content_webext_storage_local_get_ms' went from 125005 to 125005
Null values in 'scalar_parent_browser_session_restore_worker_restart_count' went from 175694 to 175694
Null values in 'normalized_channel' went from 0 to 0
Null values in 'scalar_content_telemetry_discarded_keyed_scalar_actions' went from 5811 to 5811
Null values in 'profile_reset_date' went from 130844 to 130844
...
Null values in 'scalar_content_telemetry_discarded_accumulations' went from 5811 to 5811
Datasets are not the same
```

The dropped columns may be due to some change in master since I initially ran on my branch a week ago? I'm going to look at change history.

acmiyaguchi commented:
If you look at the changelog in [`mozilla-central:/toolkit/components/telemetry/Scalars.yaml `](https://hg.mozilla.org/mozilla-central/filelog/5866d6685849311f057e7e229b9ace63a2641c29/toolkit/components/telemetry/Scalars.yaml), there are a couple related to the persistence timer and profile directory age that are consistent with the changes in the columns.
